src/Auth_Command.php

```diff
@@ -0,0 +1,332 @@
+class Auth_Command extends EE_Command {
+
+	/**
+	 * @var Filesystem $fs Symfony Filesystem object.
+	 */
+	private $fs;
+
+	/**
+	 * @var array $db Object containing essential site related information.
+	 */
+	private $db;
+
+	public function __construct() {
+
+		$this->fs = new Filesystem();
+	}
+
+
+	/**
+	 * Creates/Updates http auth for a site.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<site-name>]
+	 * : Name of website to be secured.
+	 *
+	 * [--user=<user>]
+	 * : Username for http auth.
+	 *
+	 * [--pass=<pass>]
+	 * : Password for http auth
+	 *
+	 * @alias update
+	 */
+	public function create( $args, $assoc_args ) {
+
+		$global = $this->populate_info( $args, __FUNCTION__ );
+
+		EE::debug( sprintf( 'ee auth start, Site: %s', $this->db->site_url ) );
+
+		$user = EE\Utils\get_flag_value( $assoc_args, 'user', 'easyengine' );
+		$pass = EE\Utils\get_flag_value( $assoc_args, 'pass', EE\Utils\random_password() );
+
+		EE::debug( 'Verifying htpasswd is present.' );
+		$check_htpasswd_present = EE::exec( sprintf( 'docker exec %s sh -c \'command -v htpasswd\'', EE_PROXY_TYPE ) );
+		if ( ! $check_htpasswd_present ) {
+			EE::error( sprintf( 'Could not find apache2-utils installed in %s.', EE_PROXY_TYPE ) );
+		}
+		$file   = $global ? 'default' : $this->db->site_url;
+		$params = $this->fs->exists( EE_CONF_ROOT . '/nginx/htpasswd/' . $file ) ? 'b' : 'bc';
+		EE::exec( sprintf( 'docker exec %s htpasswd -%s /etc/nginx/htpasswd/%s %s %s', EE_PROXY_TYPE, $params, $file, $user, $pass ) );
+
+		EE::log( 'Reloading global reverse proxy.' );
+		$this->reload();
+
+		EE::log( sprintf( 'Auth successfully updated for `%s` scope. New values added/updated:', $this->db->site_url ) );
+		EE::log( 'User:' . $user );
+		EE::log( 'Pass:' . $pass );
+	}
+
+	/**
+	 * Deletes http auth for a site. Default: removes http auth from site. If `--user` is passed it removes that
+	 * specific user.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<site-name>]
+	 * : Name of website.
+	 *
+	 * [--user=<user>]
+	 * : Username that needs to be deleted.
+	 */
+	public function delete( $args, $assoc_args ) {
+
+		$global = $this->populate_info( $args, __FUNCTION__ );
+		$file   = $global ? 'default' : $this->db->site_url;
+		$user   = EE\Utils\get_flag_value( $assoc_args, 'user' );
+
+		if ( $user ) {
+			EE::exec( sprintf( 'docker exec %s htpasswd -D /etc/nginx/htpasswd/%s %s', EE_PROXY_TYPE, $this->db->site_url, $user ), true, true );
+		} else {
+			$this->fs->remove( EE_CONF_ROOT . '/nginx/htpasswd/' . $file );
+			EE::log( sprintf( 'http auth removed for `%s` scope', $this->db->site_url ) );
+		}
+		$this->reload();
+	}
+
+	/**
+	 * Lists http auth users of a site.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [<site-name>]
+	 * : Name of website.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - yaml
+	 *   - json
+	 *   - count
+	 *   - text
+	 * ---
+	 */
+	public function list( $args, $assoc_args ) {
```

Comment:
I think we use __list and @subcommand for list functions.

Reply:
I guess it is time to remove that. Now that we are using min. php7, it supports the declaration of function name `list`. What should be done?

Reply:
I think it's from 7.1.0 not 7.0.
http://php.net/manual/en/function.list.php

Reply:
http://php.net/manual/en/reserved.keywords.php

>As of PHP 7.0.0 these keywords are allowed as property, constant, and method names of classes, interfaces and traits, except that class may not be used as constant name.

Reply:
ohk cool. Then we can make changes to other places. Keep it here as it is.